src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -405,6 +328,100 @@ CreateGroupedLogicalRepTargetsHash(List *logicalRepTargetList)
 }
 
 
+/*
+ * MakeTargetsCatchupWithSource uses logical replication to apply the changes
+ * made on the source to the target. It also runs all DDL on the target shards
+ * that need to be run after the data copy.
```

Comment:
I feel like this fuction does a lot more than the name suggests

Reply:
Yes, I agree. But I couldn't think of a better name.

Reply:
FinalizeShardSync

Reply:
maybe: CompleteNonBlockingShardTransfer (not in love, but more descriptive)

Reply:
I renamed it to CompleteNonBlockingShardTransfer.